core/app/models/spree/export.rb

```diff
@@ -194,5 +196,61 @@ def set_default_format
     def clear_search_params
       self.search_params = nil
     end
+
+    def normalize_date_filters(raw_params)
+      params = raw_params.is_a?(Hash) ? raw_params.deep_stringify_keys : {}
+
+      %w[created_at completed_at].each do |field|
+        ['_gt', '_gteq', '_lt', '_lteq'].each do |suffix|
+          key = "#{field}#{suffix}"
+          value = params[key]
+          next unless value.present?
+
+          params[key] = normalize_single_date_filter(value, suffix)
+        end
+      end
+
+      params
+    end
+
+    def parse_to_day_boundary(value, boundary)
+      timezone = store&.preferred_timezone.presence || Time.zone.name || 'UTC'
+
+      begin
+        date = value.respond_to?(:to_date) ? value.to_date : Date.parse(value.to_s)
+        datetime = date.in_time_zone(timezone)
+
+        case boundary
+        when :beginning_of_day
+          datetime.beginning_of_day.iso8601
+        when :end_of_day
+          datetime.end_of_day.iso8601
+        else
+          ''
+        end
+      rescue StandardError
+        ''
+      end
+    end
+
+    def normalize_single_date_filter(value, suffix)
+      if date_only?(value)
+        boundary = ['_gt', '_gteq'].include?(suffix) ? :beginning_of_day : :end_of_day
+        parse_to_day_boundary(value, boundary)
+      elsif value.is_a?(String)
+        begin
+          parsed = Time.zone.parse(value)
```

Comment:
@jawaria-sparksol this won't work properly we need to parse this in the store timezone, not the one in the rails app